hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java

```diff
@@ -197,10 +198,13 @@ public static Protocol getGrpcProtocol(Event event) {
       return Protocol.PROTOCOL_GRPC;
     }
 
-    if (event.getRpc() != null && event.getRpc().getSystem() != null) {
-      String rpcSystem = event.getRpc().getSystem();
-      if (GRPC_PROTOCOL_VALUE.equalsIgnoreCase(rpcSystem)) {
-        return Protocol.PROTOCOL_GRPC;
+    if (event.getAttributes() != null && event.getAttributes().getAttributeMap() != null) {
+      Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+      if (attributeValueMap.get(OTEL_SPAN_TAG_RPC_SYSTEM) != null) {
```

Comment:
replace this null check with attributeValueMap.containsKey(OTEL_SPAN_TAG_RPC_SYSTEM)